// @azure/arm-compute/dist/browser/operations/virtualMachineRunCommands.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { setContinuationToken } from "../pagingHelper.js";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers.js";
import * as Parameters from "../models/parameters.js";
import { createHttpPoller, } from "@azure/core-lro";
import { createLroSpec } from "../lroImpl.js";
/// <reference lib="esnext.asynciterable" />
/** Class containing VirtualMachineRunCommands operations. */
export class VirtualMachineRunCommandsImpl {
    client;
    /**
     * Initialize a new instance of the class VirtualMachineRunCommands class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Lists all available run commands for a subscription in a location.
     * @param location The name of Azure region.
     * @param options The options parameters.
     */
    list(location, options) {
        const iter = this.listPagingAll(location, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings?.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(location, options, settings);
            },
        };
    }
    async *listPagingPage(location, options, settings) {
        let result;
        let continuationToken = settings?.continuationToken;
        if (!continuationToken) {
            result = await this._list(location, options);
            let page = result.value || [];
            continuationToken = result.nextLink;
            setContinuationToken(page, continuationToken);
            yield page;
        }
        while (continuationToken) {
            result = await this._listNext(location, continuationToken, options);
            continuationToken = result.nextLink;
            let page = result.value || [];
            setContinuationToken(page, continuationToken);
            yield page;
        }
    }
    async *listPagingAll(location, options) {
        for await (const page of this.listPagingPage(location, options)) {
            yield* page;
        }
    }
    /**
     * The operation to get all run commands of a Virtual Machine.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param vmName The name of the VirtualMachine
     * @param options The options parameters.
     */
    listByVirtualMachine(resourceGroupName, vmName, options) {
        const iter = this.listByVirtualMachinePagingAll(resourceGroupName, vmName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings?.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByVirtualMachinePagingPage(resourceGroupName, vmName, options, settings);
            },
        };
    }
    async *listByVirtualMachinePagingPage(resourceGroupName, vmName, options, settings) {
        let result;
        let continuationToken = settings?.continuationToken;
        if (!continuationToken) {
            result = await this._listByVirtualMachine(resourceGroupName, vmName, options);
            let page = result.value || [];
            continuationToken = result.nextLink;
            setContinuationToken(page, continuationToken);
            yield page;
        }
        while (continuationToken) {
            result = await this._listByVirtualMachineNext(resourceGroupName, vmName, continuationToken, options);
            continuationToken = result.nextLink;
            let page = result.value || [];
            setContinuationToken(page, continuationToken);
            yield page;
        }
    }
    async *listByVirtualMachinePagingAll(resourceGroupName, vmName, options) {
        for await (const page of this.listByVirtualMachinePagingPage(resourceGroupName, vmName, options)) {
            yield* page;
        }
    }
    /**
     * Lists all available run commands for a subscription in a location.
     * @param location The name of Azure region.
     * @param options The options parameters.
     */
    _list(location, options) {
        return this.client.sendOperationRequest({ location, options }, listOperationSpec);
    }
    /**
     * Gets specific run command for a subscription in a location.
     * @param location The name of Azure region.
     * @param commandId The command id.
     * @param options The options parameters.
     */
    get(location, commandId, options) {
        return this.client.sendOperationRequest({ location, commandId, options }, getOperationSpec);
    }
    /**
     * The operation to get all run commands of a Virtual Machine.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param vmName The name of the VirtualMachine
     * @param options The options parameters.
     */
    _listByVirtualMachine(resourceGroupName, vmName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, vmName, options }, listByVirtualMachineOperationSpec);
    }
    /**
     * The operation to get the run command.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param vmName The name of the VirtualMachine
     * @param runCommandName The name of the VirtualMachineRunCommand
     * @param options The options parameters.
     */
    getByVirtualMachine(resourceGroupName, vmName, runCommandName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, vmName, runCommandName, options }, getByVirtualMachineOperationSpec);
    }
    /**
     * The operation to create or update the run command.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param vmName The name of the VirtualMachine
     * @param runCommandName The name of the VirtualMachineRunCommand
     * @param runCommand Parameters supplied to the Create Virtual Machine RunCommand operation.
     * @param options The options parameters.
     */
    async beginCreateOrUpdate(resourceGroupName, vmName, runCommandName, runCommand, options) {
        const directSendOperation = async (args, spec) => {
            return this.client.sendOperationRequest(args, spec);
        };
        const sendOperationFn = async (args, spec) => {
            let currentRawResponse = undefined;
            const providedCallback = args.options?.onResponse;
            const callback = (rawResponse, flatResponse) => {
                currentRawResponse = rawResponse;
                providedCallback?.(rawResponse, flatResponse);
            };
            const updatedArgs = {
                ...args,
                options: {
                    ...args.options,
                    onResponse: callback,
                },
            };
            const flatResponse = await directSendOperation(updatedArgs, spec);
            return {
                flatResponse,
                rawResponse: {
                    statusCode: currentRawResponse.status,
                    body: currentRawResponse.parsedBody,
                    headers: currentRawResponse.headers.toJSON(),
                },
            };
        };
        const lro = createLroSpec({
            sendOperationFn,
            args: { resourceGroupName, vmName, runCommandName, runCommand, options },
            spec: createOrUpdateOperationSpec,
        });
        const poller = await createHttpPoller(lro, {
            restoreFrom: options?.resumeFrom,
            intervalInMs: options?.updateIntervalInMs,
            resourceLocationConfig: "location",
        });
        await poller.poll();
        return poller;
    }
    /**
     * The operation to create or update the run command.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param vmName The name of the VirtualMachine
     * @param runCommandName The name of the VirtualMachineRunCommand
     * @param runCommand Parameters supplied to the Create Virtual Machine RunCommand operation.
     * @param options The options parameters.
     */
    async beginCreateOrUpdateAndWait(resourceGroupName, vmName, runCommandName, runCommand, options) {
        const poller = await this.beginCreateOrUpdate(resourceGroupName, vmName, runCommandName, runCommand, options);
        return poller.pollUntilDone();
    }
    /**
     * The operation to update the run command.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param vmName The name of the VirtualMachine
     * @param runCommandName The name of the VirtualMachineRunCommand
     * @param runCommand Parameters supplied to the Update Virtual Machine RunCommand operation.
     * @param options The options parameters.
     */
    async beginUpdate(resourceGroupName, vmName, runCommandName, runCommand, options) {
        const directSendOperation = async (args, spec) => {
            return this.client.sendOperationRequest(args, spec);
        };
        const sendOperationFn = async (args, spec) => {
            let currentRawResponse = undefined;
            const providedCallback = args.options?.onResponse;
            const callback = (rawResponse, flatResponse) => {
                currentRawResponse = rawResponse;
                providedCallback?.(rawResponse, flatResponse);
            };
            const updatedArgs = {
                ...args,
                options: {
                    ...args.options,
                    onResponse: callback,
                },
            };
            const flatResponse = await directSendOperation(updatedArgs, spec);
            return {
                flatResponse,
                rawResponse: {
                    statusCode: currentRawResponse.status,
                    body: currentRawResponse.parsedBody,
                    headers: currentRawResponse.headers.toJSON(),
                },
            };
        };
        const lro = createLroSpec({
            sendOperationFn,
            args: { resourceGroupName, vmName, runCommandName, runCommand, options },
            spec: updateOperationSpec,
        });
        const poller = await createHttpPoller(lro, {
            restoreFrom: options?.resumeFrom,
            intervalInMs: options?.updateIntervalInMs,
            resourceLocationConfig: "location",
        });
        await poller.poll();
        return poller;
    }
    /**
     * The operation to update the run command.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param vmName The name of the VirtualMachine
     * @param runCommandName The name of the VirtualMachineRunCommand
     * @param runCommand Parameters supplied to the Update Virtual Machine RunCommand operation.
     * @param options The options parameters.
     */
    async beginUpdateAndWait(resourceGroupName, vmName, runCommandName, runCommand, options) {
        const poller = await this.beginUpdate(resourceGroupName, vmName, runCommandName, runCommand, options);
        return poller.pollUntilDone();
    }
    /**
     * The operation to delete the run command.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param vmName The name of the VirtualMachine
     * @param runCommandName The name of the VirtualMachineRunCommand
     * @param options The options parameters.
     */
    async beginDelete(resourceGroupName, vmName, runCommandName, options) {
        const directSendOperation = async (args, spec) => {
            return this.client.sendOperationRequest(args, spec);
        };
        const sendOperationFn = async (args, spec) => {
            let currentRawResponse = undefined;
            const providedCallback = args.options?.onResponse;
            const callback = (rawResponse, flatResponse) => {
                currentRawResponse = rawResponse;
                providedCallback?.(rawResponse, flatResponse);
            };
            const updatedArgs = {
                ...args,
                options: {
                    ...args.options,
                    onResponse: callback,
                },
            };
            const flatResponse = await directSendOperation(updatedArgs, spec);
            return {
                flatResponse,
                rawResponse: {
                    statusCode: currentRawResponse.status,
                    body: currentRawResponse.parsedBody,
                    headers: currentRawResponse.headers.toJSON(),
                },
            };
        };
        const lro = createLroSpec({
            sendOperationFn,
            args: { resourceGroupName, vmName, runCommandName, options },
            spec: deleteOperationSpec,
        });
        const poller = await createHttpPoller(lro, {
            restoreFrom: options?.resumeFrom,
            intervalInMs: options?.updateIntervalInMs,
            resourceLocationConfig: "location",
        });
        await poller.poll();
        return poller;
    }
    /**
     * The operation to delete the run command.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param vmName The name of the VirtualMachine
     * @param runCommandName The name of the VirtualMachineRunCommand
     * @param options The options parameters.
     */
    async beginDeleteAndWait(resourceGroupName, vmName, runCommandName, options) {
        const poller = await this.beginDelete(resourceGroupName, vmName, runCommandName, options);
        return poller.pollUntilDone();
    }
    /**
     * ListNext
     * @param location The name of Azure region.
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(location, nextLink, options) {
        return this.client.sendOperationRequest({ location, nextLink, options }, listNextOperationSpec);
    }
    /**
     * ListByVirtualMachineNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param vmName The name of the VirtualMachine
     * @param nextLink The nextLink from the previous successful call to the ListByVirtualMachine method.
     * @param options The options parameters.
     */
    _listByVirtualMachineNext(resourceGroupName, vmName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, vmName, nextLink, options }, listByVirtualMachineNextOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/runCommands",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.RunCommandListResult,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.location,
    ],
    headerParameters: [Parameters.accept1],
    serializer,
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/runCommands/{commandId}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.RunCommandDocument,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.location,
        Parameters.commandId,
    ],
    headerParameters: [Parameters.accept1],
    serializer,
};
const listByVirtualMachineOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/runCommands",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.VirtualMachineRunCommandsListResult,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    queryParameters: [Parameters.apiVersion, Parameters.expand],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.vmName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const getByVirtualMachineOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/runCommands/{runCommandName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.VirtualMachineRunCommand,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    queryParameters: [Parameters.apiVersion, Parameters.expand],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.vmName,
        Parameters.runCommandName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/runCommands/{runCommandName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.VirtualMachineRunCommand,
        },
        201: {
            bodyMapper: Mappers.VirtualMachineRunCommand,
        },
        202: {
            bodyMapper: Mappers.VirtualMachineRunCommand,
        },
        204: {
            bodyMapper: Mappers.VirtualMachineRunCommand,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    requestBody: Parameters.runCommand,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.vmName,
        Parameters.runCommandName,
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer,
};
const updateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/runCommands/{runCommandName}",
    httpMethod: "PATCH",
    responses: {
        200: {
            bodyMapper: Mappers.VirtualMachineRunCommand,
            headersMapper: Mappers.VirtualMachineRunCommandsUpdateHeaders,
        },
        201: {
            bodyMapper: Mappers.VirtualMachineRunCommand,
            headersMapper: Mappers.VirtualMachineRunCommandsUpdateHeaders,
        },
        202: {
            bodyMapper: Mappers.VirtualMachineRunCommand,
            headersMapper: Mappers.VirtualMachineRunCommandsUpdateHeaders,
        },
        204: {
            bodyMapper: Mappers.VirtualMachineRunCommand,
            headersMapper: Mappers.VirtualMachineRunCommandsUpdateHeaders,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    requestBody: Parameters.runCommand1,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.vmName,
        Parameters.runCommandName,
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer,
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/runCommands/{runCommandName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        201: {},
        202: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.vmName,
        Parameters.runCommandName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.RunCommandListResult,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.subscriptionId,
        Parameters.location,
    ],
    headerParameters: [Parameters.accept1],
    serializer,
};
const listByVirtualMachineNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.VirtualMachineRunCommandsListResult,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.vmName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
