// @azure/arm-compute/dist/commonjs/operations/communityGalleries.js
"use strict";
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommunityGalleriesImpl = void 0;
const tslib_1 = require("tslib");
const coreClient = tslib_1.__importStar(require("@azure/core-client"));
const Mappers = tslib_1.__importStar(require("../models/mappers.js"));
const Parameters = tslib_1.__importStar(require("../models/parameters.js"));
/** Class containing CommunityGalleries operations. */
class CommunityGalleriesImpl {
    client;
    /**
     * Initialize a new instance of the class CommunityGalleries class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get a community gallery by gallery public name.
     * @param location The name of Azure region.
     * @param publicGalleryName The public name of the community gallery.
     * @param options The options parameters.
     */
    get(location, publicGalleryName, options) {
        return this.client.sendOperationRequest({ location, publicGalleryName, options }, getOperationSpec);
    }
}
exports.CommunityGalleriesImpl = CommunityGalleriesImpl;
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/communityGalleries/{publicGalleryName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.CommunityGallery,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    queryParameters: [Parameters.apiVersion3],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.location,
        Parameters.publicGalleryName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
