// @azure/arm-compute/dist/commonjs/operations/galleryImageVersions.js
"use strict";
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GalleryImageVersionsImpl = void 0;
const tslib_1 = require("tslib");
const pagingHelper_js_1 = require("../pagingHelper.js");
const coreClient = tslib_1.__importStar(require("@azure/core-client"));
const Mappers = tslib_1.__importStar(require("../models/mappers.js"));
const Parameters = tslib_1.__importStar(require("../models/parameters.js"));
const core_lro_1 = require("@azure/core-lro");
const lroImpl_js_1 = require("../lroImpl.js");
/// <reference lib="esnext.asynciterable" />
/** Class containing GalleryImageVersions operations. */
class GalleryImageVersionsImpl {
    client;
    /**
     * Initialize a new instance of the class GalleryImageVersions class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List gallery image versions in a gallery image definition.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param galleryName The name of the Shared Image Gallery.
     * @param galleryImageName The name of the gallery image definition to be retrieved.
     * @param options The options parameters.
     */
    listByGalleryImage(resourceGroupName, galleryName, galleryImageName, options) {
        const iter = this.listByGalleryImagePagingAll(resourceGroupName, galleryName, galleryImageName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings?.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listByGalleryImagePagingPage(resourceGroupName, galleryName, galleryImageName, options, settings);
            },
        };
    }
    async *listByGalleryImagePagingPage(resourceGroupName, galleryName, galleryImageName, options, settings) {
        let result;
        let continuationToken = settings?.continuationToken;
        if (!continuationToken) {
            result = await this._listByGalleryImage(resourceGroupName, galleryName, galleryImageName, options);
            let page = result.value || [];
            continuationToken = result.nextLink;
            (0, pagingHelper_js_1.setContinuationToken)(page, continuationToken);
            yield page;
        }
        while (continuationToken) {
            result = await this._listByGalleryImageNext(resourceGroupName, galleryName, galleryImageName, continuationToken, options);
            continuationToken = result.nextLink;
            let page = result.value || [];
            (0, pagingHelper_js_1.setContinuationToken)(page, continuationToken);
            yield page;
        }
    }
    async *listByGalleryImagePagingAll(resourceGroupName, galleryName, galleryImageName, options) {
        for await (const page of this.listByGalleryImagePagingPage(resourceGroupName, galleryName, galleryImageName, options)) {
            yield* page;
        }
    }
    /**
     * List gallery image versions in a gallery image definition.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param galleryName The name of the Shared Image Gallery.
     * @param galleryImageName The name of the gallery image definition to be retrieved.
     * @param options The options parameters.
     */
    _listByGalleryImage(resourceGroupName, galleryName, galleryImageName, options) {
        return this.client.sendOperationRequest({ resourceGroupName, galleryName, galleryImageName, options }, listByGalleryImageOperationSpec);
    }
    /**
     * Retrieves information about a gallery image version.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param galleryName The name of the Shared Image Gallery.
     * @param galleryImageName The name of the gallery image definition to be retrieved.
     * @param galleryImageVersionName The name of the gallery image version to be retrieved.
     * @param options The options parameters.
     */
    get(resourceGroupName, galleryName, galleryImageName, galleryImageVersionName, options) {
        return this.client.sendOperationRequest({
            resourceGroupName,
            galleryName,
            galleryImageName,
            galleryImageVersionName,
            options,
        }, getOperationSpec);
    }
    /**
     * Create or update a gallery image version.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param galleryName The name of the Shared Image Gallery.
     * @param galleryImageName The name of the gallery image definition to be retrieved.
     * @param galleryImageVersionName The name of the gallery image version to be retrieved.
     * @param galleryImageVersion Parameters supplied to the create or update gallery image version
     *                            operation.
     * @param options The options parameters.
     */
    async beginCreateOrUpdate(resourceGroupName, galleryName, galleryImageName, galleryImageVersionName, galleryImageVersion, options) {
        const directSendOperation = async (args, spec) => {
            return this.client.sendOperationRequest(args, spec);
        };
        const sendOperationFn = async (args, spec) => {
            let currentRawResponse = undefined;
            const providedCallback = args.options?.onResponse;
            const callback = (rawResponse, flatResponse) => {
                currentRawResponse = rawResponse;
                providedCallback?.(rawResponse, flatResponse);
            };
            const updatedArgs = {
                ...args,
                options: {
                    ...args.options,
                    onResponse: callback,
                },
            };
            const flatResponse = await directSendOperation(updatedArgs, spec);
            return {
                flatResponse,
                rawResponse: {
                    statusCode: currentRawResponse.status,
                    body: currentRawResponse.parsedBody,
                    headers: currentRawResponse.headers.toJSON(),
                },
            };
        };
        const lro = (0, lroImpl_js_1.createLroSpec)({
            sendOperationFn,
            args: {
                resourceGroupName,
                galleryName,
                galleryImageName,
                galleryImageVersionName,
                galleryImageVersion,
                options,
            },
            spec: createOrUpdateOperationSpec,
        });
        const poller = await (0, core_lro_1.createHttpPoller)(lro, {
            restoreFrom: options?.resumeFrom,
            intervalInMs: options?.updateIntervalInMs,
            resourceLocationConfig: "location",
        });
        await poller.poll();
        return poller;
    }
    /**
     * Create or update a gallery image version.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param galleryName The name of the Shared Image Gallery.
     * @param galleryImageName The name of the gallery image definition to be retrieved.
     * @param galleryImageVersionName The name of the gallery image version to be retrieved.
     * @param galleryImageVersion Parameters supplied to the create or update gallery image version
     *                            operation.
     * @param options The options parameters.
     */
    async beginCreateOrUpdateAndWait(resourceGroupName, galleryName, galleryImageName, galleryImageVersionName, galleryImageVersion, options) {
        const poller = await this.beginCreateOrUpdate(resourceGroupName, galleryName, galleryImageName, galleryImageVersionName, galleryImageVersion, options);
        return poller.pollUntilDone();
    }
    /**
     * Update a gallery image version.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param galleryName The name of the Shared Image Gallery.
     * @param galleryImageName The name of the gallery image definition to be retrieved.
     * @param galleryImageVersionName The name of the gallery image version to be retrieved.
     * @param galleryImageVersion Parameters supplied to the update gallery image version operation.
     * @param options The options parameters.
     */
    async beginUpdate(resourceGroupName, galleryName, galleryImageName, galleryImageVersionName, galleryImageVersion, options) {
        const directSendOperation = async (args, spec) => {
            return this.client.sendOperationRequest(args, spec);
        };
        const sendOperationFn = async (args, spec) => {
            let currentRawResponse = undefined;
            const providedCallback = args.options?.onResponse;
            const callback = (rawResponse, flatResponse) => {
                currentRawResponse = rawResponse;
                providedCallback?.(rawResponse, flatResponse);
            };
            const updatedArgs = {
                ...args,
                options: {
                    ...args.options,
                    onResponse: callback,
                },
            };
            const flatResponse = await directSendOperation(updatedArgs, spec);
            return {
                flatResponse,
                rawResponse: {
                    statusCode: currentRawResponse.status,
                    body: currentRawResponse.parsedBody,
                    headers: currentRawResponse.headers.toJSON(),
                },
            };
        };
        const lro = (0, lroImpl_js_1.createLroSpec)({
            sendOperationFn,
            args: {
                resourceGroupName,
                galleryName,
                galleryImageName,
                galleryImageVersionName,
                galleryImageVersion,
                options,
            },
            spec: updateOperationSpec,
        });
        const poller = await (0, core_lro_1.createHttpPoller)(lro, {
            restoreFrom: options?.resumeFrom,
            intervalInMs: options?.updateIntervalInMs,
            resourceLocationConfig: "location",
        });
        await poller.poll();
        return poller;
    }
    /**
     * Update a gallery image version.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param galleryName The name of the Shared Image Gallery.
     * @param galleryImageName The name of the gallery image definition to be retrieved.
     * @param galleryImageVersionName The name of the gallery image version to be retrieved.
     * @param galleryImageVersion Parameters supplied to the update gallery image version operation.
     * @param options The options parameters.
     */
    async beginUpdateAndWait(resourceGroupName, galleryName, galleryImageName, galleryImageVersionName, galleryImageVersion, options) {
        const poller = await this.beginUpdate(resourceGroupName, galleryName, galleryImageName, galleryImageVersionName, galleryImageVersion, options);
        return poller.pollUntilDone();
    }
    /**
     * Delete a gallery image version.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param galleryName The name of the Shared Image Gallery.
     * @param galleryImageName The name of the gallery image definition to be retrieved.
     * @param galleryImageVersionName The name of the gallery image version to be retrieved.
     * @param options The options parameters.
     */
    async beginDelete(resourceGroupName, galleryName, galleryImageName, galleryImageVersionName, options) {
        const directSendOperation = async (args, spec) => {
            return this.client.sendOperationRequest(args, spec);
        };
        const sendOperationFn = async (args, spec) => {
            let currentRawResponse = undefined;
            const providedCallback = args.options?.onResponse;
            const callback = (rawResponse, flatResponse) => {
                currentRawResponse = rawResponse;
                providedCallback?.(rawResponse, flatResponse);
            };
            const updatedArgs = {
                ...args,
                options: {
                    ...args.options,
                    onResponse: callback,
                },
            };
            const flatResponse = await directSendOperation(updatedArgs, spec);
            return {
                flatResponse,
                rawResponse: {
                    statusCode: currentRawResponse.status,
                    body: currentRawResponse.parsedBody,
                    headers: currentRawResponse.headers.toJSON(),
                },
            };
        };
        const lro = (0, lroImpl_js_1.createLroSpec)({
            sendOperationFn,
            args: {
                resourceGroupName,
                galleryName,
                galleryImageName,
                galleryImageVersionName,
                options,
            },
            spec: deleteOperationSpec,
        });
        const poller = await (0, core_lro_1.createHttpPoller)(lro, {
            restoreFrom: options?.resumeFrom,
            intervalInMs: options?.updateIntervalInMs,
            resourceLocationConfig: "location",
        });
        await poller.poll();
        return poller;
    }
    /**
     * Delete a gallery image version.
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param galleryName The name of the Shared Image Gallery.
     * @param galleryImageName The name of the gallery image definition to be retrieved.
     * @param galleryImageVersionName The name of the gallery image version to be retrieved.
     * @param options The options parameters.
     */
    async beginDeleteAndWait(resourceGroupName, galleryName, galleryImageName, galleryImageVersionName, options) {
        const poller = await this.beginDelete(resourceGroupName, galleryName, galleryImageName, galleryImageVersionName, options);
        return poller.pollUntilDone();
    }
    /**
     * ListByGalleryImageNext
     * @param resourceGroupName The name of the resource group. The name is case insensitive.
     * @param galleryName The name of the Shared Image Gallery.
     * @param galleryImageName The name of the gallery image definition to be retrieved.
     * @param nextLink The nextLink from the previous successful call to the ListByGalleryImage method.
     * @param options The options parameters.
     */
    _listByGalleryImageNext(resourceGroupName, galleryName, galleryImageName, nextLink, options) {
        return this.client.sendOperationRequest({ resourceGroupName, galleryName, galleryImageName, nextLink, options }, listByGalleryImageNextOperationSpec);
    }
}
exports.GalleryImageVersionsImpl = GalleryImageVersionsImpl;
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listByGalleryImageOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/images/{galleryImageName}/versions",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.GalleryImageVersionList,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    queryParameters: [Parameters.apiVersion3],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.galleryName,
        Parameters.galleryImageName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/images/{galleryImageName}/versions/{galleryImageVersionName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.GalleryImageVersion,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    queryParameters: [Parameters.apiVersion3, Parameters.expand12],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.galleryName,
        Parameters.galleryImageName,
        Parameters.galleryImageVersionName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const createOrUpdateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/images/{galleryImageName}/versions/{galleryImageVersionName}",
    httpMethod: "PUT",
    responses: {
        200: {
            bodyMapper: Mappers.GalleryImageVersion,
        },
        201: {
            bodyMapper: Mappers.GalleryImageVersion,
        },
        202: {
            bodyMapper: Mappers.GalleryImageVersion,
        },
        204: {
            bodyMapper: Mappers.GalleryImageVersion,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    requestBody: Parameters.galleryImageVersion,
    queryParameters: [Parameters.apiVersion3],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.galleryName,
        Parameters.galleryImageName,
        Parameters.galleryImageVersionName,
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer,
};
const updateOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/images/{galleryImageName}/versions/{galleryImageVersionName}",
    httpMethod: "PATCH",
    responses: {
        200: {
            bodyMapper: Mappers.GalleryImageVersion,
            headersMapper: Mappers.GalleryImageVersionsUpdateHeaders,
        },
        201: {
            bodyMapper: Mappers.GalleryImageVersion,
            headersMapper: Mappers.GalleryImageVersionsUpdateHeaders,
        },
        202: {
            bodyMapper: Mappers.GalleryImageVersion,
            headersMapper: Mappers.GalleryImageVersionsUpdateHeaders,
        },
        204: {
            bodyMapper: Mappers.GalleryImageVersion,
            headersMapper: Mappers.GalleryImageVersionsUpdateHeaders,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    requestBody: Parameters.galleryImageVersion1,
    queryParameters: [Parameters.apiVersion3],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.galleryName,
        Parameters.galleryImageName,
        Parameters.galleryImageVersionName,
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer,
};
const deleteOperationSpec = {
    path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/images/{galleryImageName}/versions/{galleryImageVersionName}",
    httpMethod: "DELETE",
    responses: {
        200: {},
        201: {},
        202: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    queryParameters: [Parameters.apiVersion3],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.galleryName,
        Parameters.galleryImageName,
        Parameters.galleryImageVersionName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const listByGalleryImageNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.GalleryImageVersionList,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.galleryName,
        Parameters.galleryImageName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
