// @azure/arm-compute/dist/commonjs/operations/sharedGalleryImages.js
"use strict";
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedGalleryImagesImpl = void 0;
const tslib_1 = require("tslib");
const pagingHelper_js_1 = require("../pagingHelper.js");
const coreClient = tslib_1.__importStar(require("@azure/core-client"));
const Mappers = tslib_1.__importStar(require("../models/mappers.js"));
const Parameters = tslib_1.__importStar(require("../models/parameters.js"));
/// <reference lib="esnext.asynciterable" />
/** Class containing SharedGalleryImages operations. */
class SharedGalleryImagesImpl {
    client;
    /**
     * Initialize a new instance of the class SharedGalleryImages class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * List shared gallery images by subscription id or tenant id.
     * @param location The name of Azure region.
     * @param galleryUniqueName The unique name of the Shared Gallery.
     * @param options The options parameters.
     */
    list(location, galleryUniqueName, options) {
        const iter = this.listPagingAll(location, galleryUniqueName, options);
        return {
            next() {
                return iter.next();
            },
            [Symbol.asyncIterator]() {
                return this;
            },
            byPage: (settings) => {
                if (settings?.maxPageSize) {
                    throw new Error("maxPageSize is not supported by this operation.");
                }
                return this.listPagingPage(location, galleryUniqueName, options, settings);
            },
        };
    }
    async *listPagingPage(location, galleryUniqueName, options, settings) {
        let result;
        let continuationToken = settings?.continuationToken;
        if (!continuationToken) {
            result = await this._list(location, galleryUniqueName, options);
            let page = result.value || [];
            continuationToken = result.nextLink;
            (0, pagingHelper_js_1.setContinuationToken)(page, continuationToken);
            yield page;
        }
        while (continuationToken) {
            result = await this._listNext(location, galleryUniqueName, continuationToken, options);
            continuationToken = result.nextLink;
            let page = result.value || [];
            (0, pagingHelper_js_1.setContinuationToken)(page, continuationToken);
            yield page;
        }
    }
    async *listPagingAll(location, galleryUniqueName, options) {
        for await (const page of this.listPagingPage(location, galleryUniqueName, options)) {
            yield* page;
        }
    }
    /**
     * List shared gallery images by subscription id or tenant id.
     * @param location The name of Azure region.
     * @param galleryUniqueName The unique name of the Shared Gallery.
     * @param options The options parameters.
     */
    _list(location, galleryUniqueName, options) {
        return this.client.sendOperationRequest({ location, galleryUniqueName, options }, listOperationSpec);
    }
    /**
     * Get a shared gallery image by subscription id or tenant id.
     * @param location The name of Azure region.
     * @param galleryUniqueName The unique name of the Shared Gallery.
     * @param galleryImageName The name of the Shared Gallery Image Definition from which the Image
     *                         Versions are to be listed.
     * @param options The options parameters.
     */
    get(location, galleryUniqueName, galleryImageName, options) {
        return this.client.sendOperationRequest({ location, galleryUniqueName, galleryImageName, options }, getOperationSpec);
    }
    /**
     * ListNext
     * @param location The name of Azure region.
     * @param galleryUniqueName The unique name of the Shared Gallery.
     * @param nextLink The nextLink from the previous successful call to the List method.
     * @param options The options parameters.
     */
    _listNext(location, galleryUniqueName, nextLink, options) {
        return this.client.sendOperationRequest({ location, galleryUniqueName, nextLink, options }, listNextOperationSpec);
    }
}
exports.SharedGalleryImagesImpl = SharedGalleryImagesImpl;
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const listOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/sharedGalleries/{galleryUniqueName}/images",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SharedGalleryImageList,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    queryParameters: [Parameters.apiVersion3, Parameters.sharedTo],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.location,
        Parameters.galleryUniqueName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const getOperationSpec = {
    path: "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/sharedGalleries/{galleryUniqueName}/images/{galleryImageName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SharedGalleryImage,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    queryParameters: [Parameters.apiVersion3],
    urlParameters: [
        Parameters.$host,
        Parameters.subscriptionId,
        Parameters.location,
        Parameters.galleryImageName,
        Parameters.galleryUniqueName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
const listNextOperationSpec = {
    path: "{nextLink}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SharedGalleryImageList,
        },
        default: {
            bodyMapper: Mappers.CloudError,
        },
    },
    urlParameters: [
        Parameters.$host,
        Parameters.nextLink,
        Parameters.subscriptionId,
        Parameters.location,
        Parameters.galleryUniqueName,
    ],
    headerParameters: [Parameters.accept],
    serializer,
};
